docs/apache-airflow/upgrading-to-2.rst

```diff
@@ -0,0 +1,1164 @@
+
+.. code-block:: python
+
+    second_task = PythonOperator(
+        task_id="four_task",
+        python_callable=test_volume_mount,
+        executor_config={
+            "KubernetesExecutor": {
+                "volumes": [
+                    {
+                        "name": "example-kubernetes-test-volume",
+                        "hostPath": {"path": "/tmp/"},
+                    },
+                ],
+                "volume_mounts": [
+                    {
+                        "mountPath": "/foo/",
+                        "name": "example-kubernetes-test-volume",
+                    },
+                ]
+            }
+        }
+    )
+
+In the new model a user can accomplish the same thing using the following code under the ``pod_override`` key:
+
+.. code-block:: python
+
+    from kubernetes.client import models as k8s
+
+    second_task = PythonOperator(
+        task_id="four_task",
+        python_callable=test_volume_mount,
+        executor_config={"pod_override": k8s.V1Pod(
+            spec=k8s.V1PodSpec(
+                containers=[
+                    k8s.V1Container(
+                        name="base",
+                        volume_mounts=[
+                            k8s.V1VolumeMount(
+                                mount_path="/foo/",
+                                name="example-kubernetes-test-volume"
+                            )
+                        ]
+                    )
+                ],
+                volumes=[
+                    k8s.V1Volume(
+                        name="example-kubernetes-test-volume",
+                        host_path=k8s.V1HostPathVolumeSource(
+                            path="/tmp/"
+                        )
+                    )
+                ]
+            )
+        )
+        }
+    )
+
+For Airflow 2.0, the traditional ``executor_config`` will continue operation with a deprecation warning,
+but will be removed in a future version.
+
+
+Step 7: Upgrade to Airflow 2.0
+'''''''''''''''''''''''''''''''
+
+After running the upgrade checks as described above, installing the backported providers, modifying
+the DAGs to be compatible, and updating the configuration settings, you should be ready to upgrade to Airflow 2.0.
+
+A final run of the upgrade checks is always a good idea to make sure you have missed anything. At
+this stage the problems detected should be either be zero or minimal which you plan to fix after
+upgrading the Airflow version.
+
+At this point, just follow the standard Airflow version upgrade process:
+* Make sure your Airflow meta database is backed up
+* Pause all the DAGs and make sure there is nothing actively running
+* Upgrade the Airflow version to the 2.0 version of choice
+* Upgrade the Airflow meta database using ``airflow db upgrade``. This command is now shown in the Airflow 2.0 CLI syntax.
+* Restart Airflow Scheduler, Webserver, and Workers
```

Comment:
I think we should explicitly mention something about installing the right providers, either via the extras or selecting the (non backport) providers too.

Possibly have to _uninstall_ the backport providers too if you are upgrading an install in place (i.e. if you are not using Docker where a change to requirements gets a fresh set of modules)